src/geometry/point.rs

```diff
@@ -232,6 +232,27 @@ impl Point {
     pub fn component_div(self, other: Self) -> Self {
         Self::new(self.x / other.x, self.y / other.y)
     }
+
+    /// Returns a point that is rotated by 90° relative to the origin.
+    pub(crate) const fn rotate_90(self) -> Self {
+        Self::new(self.y, -self.x)
+    }
+
+    /// Calculates the dot product of two points.
+    pub(crate) const fn dot_product(self, other: Point) -> i32 {
+        self.x * other.x + self.y * other.y
+    }
+
+    /// Calculates the determinant of a 2x2 matrix formed by this and another point.
+    ///
+    /// ```text
+    ///          | self.x  self.y  |
+    /// result = |                 |
+    ///          | other.x other.y |
+    ///
```

Comment:
Not strictly necessary but good for consistency.

```suggestion
    /// ```
```